plugins/languages/markdown/markview.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "markview";
+  originalName = "markview.nvim";
+  defaultPackage = pkgs.vimPlugins.markview-nvim;
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  description = ''
+    An experimental markdown previewer for Neovim.
+
+    Supports a vast amount of rendering customization.
+    Please refer to the plugin's [documentation](https://github.com/OXY2DEV/markview.nvim/wiki/Configuration-options) for more details.
+  '';
+
+  # TODO: remove when https://github.com/NixOS/nixpkgs/pull/333587 is available
+  extraPlugins = with pkgs.vimPlugins; [ nvim-web-devicons ];
+
+  settingsOptions = {
+    buf_ignore = defaultNullOpts.mkListOf types.str [ "nofile" ] ''
+      Buftypes to disable markview-nvim.
+    '';
+
+    mode = defaultNullOpts.mkListOf types.str [
+      "n"
+      "no"
+    ] "Modes where preview is enabled.";
+
+    hybrid_modes = defaultNullOpts.mkListOf types.str null "Modes where hybrid mode is enabled.";
+
+    callback = {
+      on_enable = defaultNullOpts.mkLuaFn' {
+        pluginDefault = # Lua
+          ''
+            function(buf, win)
+              vim.wo[window].conceallevel = 2;
+              vim.wo[window].concealcursor = "nc";
+            end
+          '';
+        description = ''
+          Action to perform when markview is enabled.
+        '';
+      };
+
+      on_disable = defaultNullOpts.mkLuaFn' {
+        pluginDefault = # Lua
+          ''
+            function(buf, win)
+                vim.wo[window].conceallevel = 0;
+                vim.wo[window].concealcursor = "";
+            end
+          '';
```

Comment:
```suggestion
        pluginDefault = ''
          function(buf, win)
              vim.wo[window].conceallevel = 0;
              vim.wo[window].concealcursor = "";
          end
        '';
```

No need to specify that it is lua.

Reply:
This is for syntax highlighting since we're not using .__raw anymore.

Reply:
Ok ! And this comment actually triggers syntax highlighting for lua ?

Reply:
![image](https://github.com/user-attachments/assets/3e202abf-825f-4748-869b-b16d774ce4f4)
vs 
![image](https://github.com/user-attachments/assets/6a00908c-c7a3-4e9c-b531-84ef6f5b97f8)

Reply:
Ok, I'm fine with leaving them then.